packages/stark-core/src/common/bootstrap/abstract-stark-main.ts

```diff
@@ -0,0 +1,157 @@
+		// no need for the event listener after this
+		document.removeEventListener("DOMContentLoaded", this.mainWrapper, false);
+
+		if (this.environment.hmr) {
+			if (module["hot"]) {
+				this.bootstrapHmr(module, this.main);
+			} else {
+				console.error("HMR is not enabled for webpack-dev-server!");
+			}
+		} else {
+			this.invokeMain();
+		}
+	};
+	
+	private invokeMain():void {
+		// FIXME handle errors nicely (UX!)
+		this.main().catch((err: any) => console.error(err));
+
+		// TODO reintroduce later
+		// bootstrap the app as long as the app root element is there (it won't be there in case the browser is outdated)
+		// if (document.getElementsByClassName("stark-app")[0]) {
+		// 	// add anything relevant here
+		// } else {
+		// 	console.error("Could not bootstrap the App: unsupported browser.");
+		// }
+	}
+
+	/**
+	 * Calls the main function once the DOM has loaded
+	 */
+	protected bootstrapDomReady(): void {
+		switch (document.readyState) {
+			case "loading":
+				document.addEventListener("DOMContentLoaded", this.mainWrapper, false);
+				break;
+			case "interactive":
+			case "complete":
+			default:
+				if (this.environment.hmr) {
+					if (module["hot"]) {
+						this.bootstrapHmr(module, this.main);
+					} else {
+						console.error("HMR is not enabled for webpack-dev-server! This is most probably due to a bug in Stark.");
+					}
+				} else {
+					this.invokeMain();
+				}
+		}
+	}
+
+	/**
+	 * Bootstrap after document is ready
+	 */
+	public bootstrap(): void {
+		console.log(`
+NNNNNNNN        NNNNNNNNBBBBBBBBBBBBBBBBB   BBBBBBBBBBBBBBBBB   
+N:::::::N       N::::::NB::::::::::::::::B  B::::::::::::::::B
+N::::::::N      N::::::NB::::::BBBBBB:::::B B::::::BBBBBB:::::B
+N:::::::::N     N::::::NBB:::::B     B:::::BBB:::::B     B:::::B
+N::::::::::N    N::::::N  B::::B     B:::::B  B::::B     B:::::B
+N:::::::::::N   N::::::N  B::::B     B:::::B  B::::B     B:::::B
+N:::::::N::::N  N::::::N  B::::BBBBBB:::::B   B::::BBBBBB:::::B
+N::::::N N::::N N::::::N  B:::::::::::::BB    B:::::::::::::BB
+N::::::N  N::::N:::::::N  B::::BBBBBB:::::B   B::::BBBBBB:::::B
+N::::::N   N:::::::::::N  B::::B     B:::::B  B::::B     B:::::B
+N::::::N    N::::::::::N  B::::B     B:::::B  B::::B     B:::::B
+N::::::N     N:::::::::N  B::::B     B:::::B  B::::B     B:::::B
+N::::::N      N::::::::NBB:::::BBBBBB::::::BBB:::::BBBBBB::::::B
+N::::::N       N:::::::NB:::::::::::::::::B B:::::::::::::::::B
+N::::::N        N::::::NB::::::::::::::::B  B::::::::::::::::B
+NNNNNNNN         NNNNNNNBBBBBBBBBBBBBBBBB   BBBBBBBBBBBBBBBBB
+
+We need great software developers like you! https://jobs.nbb.be
+		`);
+
+		this.bootstrapDomReady();
+	}
+
+	/**
+	 * Configure HMR
+	 * Code based on: https://github.com/angular/angular-cli/wiki/stories-configure-hmr
+	 * Reference: https://github.com/gdi2290/angular-hmr
+	 * @ignore
+	 */
+	protected bootstrapHmr: any = (module: any, bootstrap: () => Promise<NgModuleRef<any>>) => {
```

Comment:
Maybe you can change the type of this to Function ;)

Reply:
Yup!